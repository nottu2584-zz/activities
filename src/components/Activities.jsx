import { default as React } from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  activities: {
    background: "#F3F4F6",
    color: "#282c34",
    borderRadius: 5,
    fontSize: "0.9em",
    overflow: "hidden",
    padding: 0,
  },
  row: {
    display: "flex",
    flexWrap: "wrap",
    width: "100%",
    "&.head": {
      borderBottom: "1px solid #DCDFE5",
      fontFamily: "'Rubik', sans-serif",
      fontSize: "1.2em",
      fontWeight: "bold",
      textTransform: "uppercase",
    },
    "&:nth-child(even)": {
      backgroundColor: "#DCDFE5",
    },
  },
  cell: {
    boxSizing: "border-box",
    flexGrow: "1",
    padding: "0.8em 1.2em",
    overflow: "hidden",
    listStyle: "none",
    "&.col-1": {
      flex: "0 0 50px",
    },
    "&.col-2": {
      flex: "0 0 calc(50% - 50px)",
      textAlign: "left",
    },
    "&.col-3": {
      flex: "0 0 25%",
    },
    "&.col-4": {
      flex: "0 0 25%",
    },
  },
});

const Activities = (props) => {
  const classes = useStyles();
  const { activities, checked, handleOnChange } = props;

  return (
    <div className={`activities ${classes.activities}`}>
      <div className={`${classes.row} row head`}>
        <div className={`${classes.cell} col col-1`}></div>
        <div className={`${classes.cell} col col-2`}>Activity Name</div>
        <div className={`${classes.cell} col col-3`}>COVID Friendly</div>
        <div className={`${classes.cell} col col-4`}>Times Performed</div>
      </div>
      {activities
        ? activities.map((activity, index) => {
            return (
              <div key={index} className={`${classes.row} row row-${index}`}>
                <div className={`${classes.cell} col col-1`}>
                  <input
                    name={`activity-${activity.id}`}
                    id={`activity-${activity.id}`}
                    type="checkbox"
                    checked={checked.some(
                      (item) => item.id === activity.id
                    )}
                    onChange={(e) => handleOnChange(e, activity)}
                  />
                </div>
                <div className={`${classes.cell} col col-2`}>
                  {activity.name}
                </div>
                <div className={`${classes.cell} col col-3`}>
                  {activity.friendly ? "Yes" : "No"}
                </div>
                <div className={`${classes.cell} col col-4`}>
                  {activity.performed}
                </div>
              </div>
            );
          })
        : null}
    </div>
  );
};

export default Activities;
