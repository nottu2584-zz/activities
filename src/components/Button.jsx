
import { default as React } from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  button: {
    background: "#009b72",
    boxShadow: "0 1px 2px #333",
    border: "none",
    borderRadius: "5px",
    color: "#ffffff",
    cursor: "pointer",
    fontFamily: "'Rubik', sans-serif",
    fontSize: "1.1em",
    lineHeight: "1.1em",
    outline: "none",
    padding: "1em 1.25em",
    MozTransition: "box-shadow 0.25s",
    OTransition: "box-shadow 0.25s",
    WebkitTransition: "box-shadow 0.25s",
    transition: "box-shadow 0.25s",
    WebkitFontSmoothing: "antialiased",
    MozOsxFontSmoothing: "grayscale",
  },
  "&:hover": { boxShadow: "0 1px 20px #333" },
});

const Button = (props) => {
  const classes = useStyles();
  
  const {handleOnClick} = props;

  return (
    <button className={classes.button} onClick={(e) => handleOnClick(e)}>
      {props.children}
    </button>
  );
};

export default Button;