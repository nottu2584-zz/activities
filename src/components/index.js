export { default as Activities } from "./Activities";
export { default as Button } from "./Button";
export { default as Modal } from "./Modal";

