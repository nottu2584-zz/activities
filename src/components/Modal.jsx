import { createUseStyles } from "react-jss";
import Button from "./Button";

const useStyles = createUseStyles({
  modal: {
    backgroundColor: "rgba(0, 0, 0, 0.85)",
    position: "fixed",
    height: "100%",
    width: "100%",
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
    zIndex: 1,
  },
  wrapper: {
    display: "flex",
    flexDirection: "column",
    flexWrap: "wrap",
    alignItems: "center",
    height: "100%",
    justifyContent: "center",
    zIndex: 2,
  },
  content: {
    display: "inline-flex",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    maxWidth: "30%",
    textAlign: "center",
    "& p": {
      margin: "1em 0 0",
    },
    "& ul": {
      textAlign: "left",
      margin: "1em 0",
    },
  },
  fullWidth: {
    flex: "1 1 100%",
  },
  button: {
    background: "#009B72",
    boxShadow: "0 1px 2px #333",
    border: "none",
    borderRadius: 5,
    color: "#FFFFFF",
    cursor: "pointer",
    fontSize: "1.25em",
    lineHeight: "1.1em",
    outline: "none",
    padding: "1em 1.25em",
    "-moz-transition": "box-shadow 0.25s",
    "-o-transition": "box-shadow 0.25s",
    "-webkit-transition": "box-shadow 0.25s",
    transition: "box-shadow 0.25s",
    "&:hover": {
      boxShadow: "0 1px 20px #333",
    },
  },
});

const Modal = (props) => {
  const classes = useStyles();

  const { activities, handleModal, message, title } = props;

  return (
    <div className={`modal ${classes.modal}`}>
      <div className={`wrapper ${classes.wrapper}`}>
        <div className={`content ${classes.content}`}>
          <h2 className={classes.fullWidth}>{title}</h2>
          <p className={classes.fullWidth}>{message}</p>
          <ul className={classes.fullWidth}>
            {activities
              ? activities.map((activity, index) => (
                  <li key={index}>{activity.name}</li>
                ))
              : null}
          </ul>
          <Button handleOnClick={handleModal}>Oh, I see!</Button>
        </div>
      </div>
    </div>
  );
};

export default Modal;
