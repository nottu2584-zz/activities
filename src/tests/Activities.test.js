import { shallow } from "enzyme";
import { Activities } from "../components";

describe("<Activities /> without props", () => {
  const container = shallow(<Activities />);
  it("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  it("should have an activities wrapper", () => {
    expect(container.find(".activities").length).toEqual(1);
  });

  it("should have a single head row", () => {
    expect(container.find(".head").length).toEqual(1);
  });

  it("should have at least four columns", () => {
    expect(container.find(".col").length).toBeGreaterThanOrEqual(4);
  });
});

describe("<Activities /> with props", () => {
  const initialProps = {
    activities: [
      {
        id: 1,
        name: "Netflix & Chill",
        friendly: true,
        performed: 50,
      },
      {
        id: 2,
        name: "Birthdate in a restaurant",
        friendly: false,
        performed: 0,
      },
      {
        id: 3,
        name: "Vacations in Hawaii",
        friendly: false,
        performed: 1,
      },
      {
        id: 4,
        name: "Becoming Maria Kondo",
        friendly: true,
        performed: 2,
      },
    ],
    checked: [],
    handleOnChange: jest.fn(),
  };
  const container = shallow(<Activities {...initialProps} />);

  it("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  it("should have an activities wrapper", () => {
    expect(container.find(".activities").length).toEqual(1);
  });
  
  it("should have a single head row", () => {
    expect(container.find(".head").length).toEqual(1);
  });

  it("should have at least eight cells", () => {
    expect(container.find(".col").length).toBeGreaterThanOrEqual(8);
  });

  it("should have checkboxes unchecked", () => {
    container.find('input[type="checkbox"]').forEach((node) => {
      expect(node.props().checked).toEqual(false);
    });
  });

  it("should have some checkboxes unchecked and a checkbox checked", () => {
    initialProps.checked = [
      {
        id: 3,
        name: "Vacations in Hawaii",
        friendly: false,
        performed: 1,
      },
    ];

    container.find('input[type="checkbox"]').forEach((node, index) => {
      if (initialProps.activities[index] === 3)
        expect(node.props().checked).toEqual(true);
      else expect(node.props().checked).toEqual(false);
    });
  });

  it("should handle an activity in an onChange event", () => {
    const event = {
      preventDefault() {},
    };

    container.find('input[type="checkbox"]').forEach((node, index) => {
      node.simulate("change", event);
      expect(initialProps.handleOnChange).toBeCalledWith(
        event,
        initialProps.activities[index]
      );
    });
  });
});
