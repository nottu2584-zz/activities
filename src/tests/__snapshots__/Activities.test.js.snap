// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activities /> with props should match the snapshot 1`] = `"<div class=\\"activities activities-0-2-1\\"><div class=\\"row-0-2-2 row head\\"><div class=\\"cell-0-2-3 col col-1\\"></div><div class=\\"cell-0-2-3 col col-2\\">Activity Name</div><div class=\\"cell-0-2-3 col col-3\\">COVID Friendly</div><div class=\\"cell-0-2-3 col col-4\\">Times Performed</div></div><div class=\\"row-0-2-2 row row-0\\"><div class=\\"cell-0-2-3 col col-1\\"><input type=\\"checkbox\\" name=\\"activity-1\\" id=\\"activity-1\\"/></div><div class=\\"cell-0-2-3 col col-2\\">Netflix &amp; Chill</div><div class=\\"cell-0-2-3 col col-3\\">Yes</div><div class=\\"cell-0-2-3 col col-4\\">50</div></div><div class=\\"row-0-2-2 row row-1\\"><div class=\\"cell-0-2-3 col col-1\\"><input type=\\"checkbox\\" name=\\"activity-2\\" id=\\"activity-2\\"/></div><div class=\\"cell-0-2-3 col col-2\\">Birthdate in a restaurant</div><div class=\\"cell-0-2-3 col col-3\\">No</div><div class=\\"cell-0-2-3 col col-4\\">0</div></div><div class=\\"row-0-2-2 row row-2\\"><div class=\\"cell-0-2-3 col col-1\\"><input type=\\"checkbox\\" name=\\"activity-3\\" id=\\"activity-3\\"/></div><div class=\\"cell-0-2-3 col col-2\\">Vacations in Hawaii</div><div class=\\"cell-0-2-3 col col-3\\">No</div><div class=\\"cell-0-2-3 col col-4\\">1</div></div><div class=\\"row-0-2-2 row row-3\\"><div class=\\"cell-0-2-3 col col-1\\"><input type=\\"checkbox\\" name=\\"activity-4\\" id=\\"activity-4\\"/></div><div class=\\"cell-0-2-3 col col-2\\">Becoming Maria Kondo</div><div class=\\"cell-0-2-3 col col-3\\">Yes</div><div class=\\"cell-0-2-3 col col-4\\">2</div></div></div>"`;

exports[`<Activities /> without props should match the snapshot 1`] = `"<div class=\\"activities activities-0-2-1\\"><div class=\\"row-0-2-2 row head\\"><div class=\\"cell-0-2-3 col col-1\\"></div><div class=\\"cell-0-2-3 col col-2\\">Activity Name</div><div class=\\"cell-0-2-3 col col-3\\">COVID Friendly</div><div class=\\"cell-0-2-3 col col-4\\">Times Performed</div></div></div>"`;
